hack/ci/e2e-k8s.sh

```diff
@@ -24,7 +24,12 @@ set -o errexit -o nounset -o xtrace
 # FOCUS: ginkgo focus regex
 # GA_ONLY: true  - limit to GA APIs/features as much as possible
 #          false - (default) APIs and features left at defaults
-# 
+# FEATURE_GATES:
```

Comment:
I think maybe we should just require a valid yaml map string in the env and not try to do any bash parsing. Same for runtime_config

like `FEATURE_GATES='{foo: true}'`

Reply:
Makes sense, will change it.

Reply:
I keep forgetting that YAML supports JSON and thus everything in a single string with no line breaks :sweat_smile:

Reply:
There's one small (and perfectly acceptable) loss of functionality when removing the parsing:

> For the sake of simplicity, not attempt is made to merge the settings from
> GA_ONLY with these new env variables. If GA_ONLY=true is set, the new env
> variables are ignored.

Revised commit pushed.

/hold

I still need to test it.

Reply:
I tested:
- no vars set
- GA_ONLY=true
- FEATURE_GATES/RUNTIME_CONFIG as in https://github.com/kubernetes/test-infra/pull/28248
- all variables

That all works as expected now.

/hold cancel

Reply:
The script is also being used in https://testgrid.k8s.io/sig-instrumentation-tests#kind-json-logging-master through https://github.com/pohly/test-infra/blob/cc00ee53e0bee4a32e3171d86ddfc67438c81219/config/jobs/kubernetes/sig-instrumentation/sig-instrumentation-kind-periodics.yaml#L36-L43

That also works.